### PR TITLE
Universal Kafka docs

### PR DESCRIPTION
**Pull request by jzdr**

### CODE REVIEW

File: docs/scenarios_authoring/DataSourcesAndSinks.md [hunk 1]
```diff
@@ -9,145 +9,29 @@ sidebar_position: 3
...
-Another aspect differentiating source and sink components is data serialization. Data serialization determines how data fields of input JSON objects are physically encoded in the “series of bytes” that are put into the Kafka topic as event data. Two main methods are used:
+## Kafka source
 
+Nussknacker will connect to Schema Registry to get the list of available schemas and will pre-populate the `Topic name` drop down of the kafka-avro component with the list of available topic names.
 
+Nussknacker Designer knows what data to expect in the Kafka event so it will hint field names and will perform additional validations on SpEL expressions involving data from the selected topic. You can select the schema version which should be used for that.
 
-* Avro serialization
-* JSON serialization
+![Kafka source](img/kafkaAvroSource.png "Kafka source")
```
Comment (gadomsky) [inline docs/scenarios_authoring/DataSourcesAndSinks.md:29]: `img/kafkaSource.png`

File: docs/scenarios_authoring/DataSourcesAndSinks.md [hunk 2]
```diff
@@ -9,145 +9,29 @@ sidebar_position: 3
...
+It is not uncommon that  programs that write events to Kafka topics (“producers”) and programs that read events from Kafka topics (“consumers”) are implemented by different people. If consumers are to understand what producers put into the topic, they need to agree between themselves the data schema - the format and data types of the data transported over Kafka topics. This “contract” is kept in what is called a Schema Registry. Nussnacker is able to read it and use to determine what data is in the event and help with field names and data type related validation of SpEL expressions.
 
-Another aspect differentiating source and sink components is data serialization. Data serialization determines how data fields of input JSON objects are physically encoded in the “series of bytes” that are put into the Kafka topic as event data. Two main methods are used:
+## Kafka source
 
+Nussknacker will connect to Schema Registry to get the list of available schemas and will pre-populate the `Topic name` drop down of the kafka-avro component with the list of available topic names.
```
Comment (gadomsky) [inline docs/scenarios_authoring/DataSourcesAndSinks.md:25]: `kafka-avro` ?

File: docs/scenarios_authoring/DataSourcesAndSinks.md [hunk 3]
```diff
@@ -9,145 +9,29 @@ sidebar_position: 3
...
 
+Nussknacker Designer knows what data to expect in the Kafka event so it will hint field names and will perform additional validations on SpEL expressions involving data from the selected topic. You can select the schema version which should be used for that.
```
Comment (gadomsky) [inline docs/scenarios_authoring/DataSourcesAndSinks.md:27]: Let's be consistent - in previous paragraph we use term `Nussknacker`

File: docs/scenarios_authoring/DataSourcesAndSinks.md [hunk 4]
```diff
@@ -9,145 +9,29 @@ sidebar_position: 3
...
 
-* Avro serialization
-* JSON serialization
+![Kafka source](img/kafkaAvroSource.png "Kafka source")
 
-Therefore, it is important to understand how the data in the incoming Kafka topic are serialized; similarly, how data should be serialized in the result (output) Kafka topic. Serialization is relevant only for source and sink components. 
+## Kafka sink
 
-Below, source and sink components are presented in groups with identical approaches to data typing and serialization. The table summarizing different types of source and sink components is at the end of this page. 
+The kafka sink configuration form will show a list of fields defined in Schema Registry for the given topic. The result of the expression entered in the `Key` field will be used as a partitioning key when sending the record to Kafka.
```
Comment (gadomsky) [inline docs/scenarios_authoring/DataSourcesAndSinks.md:33]: with capital letter? Kafka?

File: docs/scenarios_authoring/DataSourcesAndSinks.md [hunk 5]
```diff
@@ -9,145 +9,29 @@ sidebar_position: 3
...
-| Schema Registry + JSON serialization     | Avro in Confluent Schema Registry           | JSON                       | Avro supported               | Slightly more involving initially because of Schema Registry                     | <p> Hints and data type related validations </p> Maintainability  |
-| Manually typed + JSON serialization      | Declared inside  Kafka-typed-json component | JSON                       | Common between Java and JSON | <p> May become difficult to maintain </p> No validation of input data at runtime | Hints during authoring                                            |
-| No type information + JSON serialization | none                                        | JSON                       | Common between Java and JSON | No hints and data type related validations                                       | Good for JSON objects whose structure is not known at design time |
-
- 
-
+If you prefer to define the whole output event as one expression containing value to be written to the Kafka topic, instead of filling separete autogenerated fields, you can do that by switching `Raw editor` to `true`.
```
Comment (gadomsky) [inline docs/scenarios_authoring/DataSourcesAndSinks.md:37]: When using `Raw editor = true` one should also choose `Validation mode`. Maybe we should describe it too.

strict - Requires providing all fields, including optional fields, without redundant fields
lax - Requires providing only required fields, without optional fields, with redundant fields

File: docs/scenarios_authoring/DataSourcesAndSinks.md (hunk 2 above)
Comment (jzdr) [reply docs/scenarios_authoring/DataSourcesAndSinks.md:25]: fixed

File: docs/scenarios_authoring/DataSourcesAndSinks.md (hunk 3 above)
Comment (jzdr) [reply docs/scenarios_authoring/DataSourcesAndSinks.md:27]: Changed to just Nussknacker.

File: docs/scenarios_authoring/DataSourcesAndSinks.md (hunk 1 above)
Comment (jzdr) [reply docs/scenarios_authoring/DataSourcesAndSinks.md:29]: fixed

File: docs/scenarios_authoring/DataSourcesAndSinks.md (hunk 4 above)
Comment (jzdr) [reply docs/scenarios_authoring/DataSourcesAndSinks.md:33]: Changed kafka to `kafka`

File: docs/scenarios_authoring/DataSourcesAndSinks.md (hunk 5 above)
Comment (jzdr) [reply docs/scenarios_authoring/DataSourcesAndSinks.md:37]: Added

**Review by dswiecki — APPROVED** [review]

**Pull request merged**